Cleanup services

Allcharles opened the pull request:
# Cleanup Services

Cleanup the current services in preparation for creating all remaining services.

## Changes

- Redesign of service unit testing
- Enabled the following mocked services:
  - `TagsService`
  - `ScriptsService`

## Issues

- New test suite does not provide testing for additional parameters (ie. Project model in sites services). It only tests the parameter as a number or string value instead of allowing an AbstractModel input

Allcharles committed:
Re-enabled tags service

atruskie commented on src/app/services/baw-api/api-common.ts:
```diff
@@ -4,28 +4,54 @@ import { Param } from "src/app/interfaces/apiInterfaces";
+
+/**
+ * Convert baw model or id to optional parameter
+ */
```
Type alias are not really converting anything, they're just saying this type can be x or y (in this case)

Allcharles committed:
Cleanup simple service tests

Allcharles committed:
Cleanup sites service tests

Allcharles committed:
Re-enabled scripts service

Cleanup review comment: https://github.com/QutEcoacoustics/workbench-client/pull/224#discussion_r413639569

Allcharles committed:
Implemented retrieveResolvers tests

Allcharles committed:
Enable security service authTrigger tests

Allcharles committed:
Created immutable api mock

atruskie commented on src/app/services/baw-api/api-common.helper.ts:
```diff
@@ -0,0 +1,148 @@
+import { Id } from "@interfaces/apiInterfaces";
```
is this a test-support file?

All of the validate* functions seem to only be usable in tests. If that's the case we should indicated the intended use for this file. I'd move it to `tests/helpers` or something like that.

atruskie reviewed (approved):
I appreciate what you've done with the service tests. A huge amount of boilerplate testing removed and just testing the common class. That's great.

Looks good,

Allcharles committed:
Extracted all test functions to folder
